### PR TITLE
skeleton implementation of rpm builder

### PR DESCRIPTION
**Pull request by AnderEnder**
Initial draft implementation of rpm builder.

### CODE REVIEW

File: src/header/mod.rs [hunk 1]
```diff
@@ -40,6 +44,99 @@ where
         }
     }
 
+    pub fn insert(&mut self, key: T, value: RType) {
```
Comment (andy128k) [inline src/header/mod.rs]: Consider `... ) -> &mut Self {` to enable chaining.

File: src/rpm/builder.rs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+use super::info::RPMInfo;
+
+struct InnerPath {
+    path: String,
+    user: String,
+    group: String,
+    mode: u8,
+}
+
+#[derive(Debug, Default)]
```
Comment (andy128k) [inline src/rpm/builder.rs]: Default enables bad code like `RPMBuilder::default().build()`.

File: src/rpm/builder.rs [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    // postTrans
+
+    pub fn add_file(mut self, file: &str) -> Self {
+        self.files.push(file.to_owned());
+        self
+    }
+
+    pub fn add_files(mut self, files: Vec<&str>) -> Self {
+        let files_vec: Vec<String> = files.into_iter().map(|x| x.to_owned()).collect();
+        self.files.extend_from_slice(&files_vec);
```
Comment (andy128k) [inline src/rpm/builder.rs]: Probably a loop will be better than intermediate allocation of `files_vec`.

File: src/rpm/builder.rs [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        self
+    }
+
+    pub fn add_links(mut self, dirs: Vec<&str>) -> Self {
+        let links_vec: Vec<String> = dirs.into_iter().map(|x| x.to_owned()).collect();
+        self.links.extend_from_slice(&links_vec);
+        self
+    }
+
+    pub fn build(self) {
```
Comment (andy128k) [inline src/rpm/builder.rs]: `-> io::Result<()>`

File: src/rpm/builder.rs (hunk 2 above)
Comment (AnderEnder) [reply src/rpm/builder.rs]: Good point. It was used to fast start. It should be removed by the end of this PR.

File: src/rpm/builder.rs (hunk 3 above)
Comment (AnderEnder) [reply src/rpm/builder.rs]: Agree. It even looks quite ugly.

File: src/rpm/builder.rs (hunk 2 above)
Comment (AnderEnder) [reply src/rpm/builder.rs]: @andy128k,
According to `builder_derive`, they uses exactly this pattern:  `RPMBuilder::default().something().build()`

**Pull request merged**